test-framework/junit5-component/src/test/java/io/quarkus/test/component/ProgrammaticLookupMockTest.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.test.component;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import jakarta.enterprise.inject.Instance;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.Mockito;
+
+import io.quarkus.test.component.beans.Delta;
+
+public class ProgrammaticLookupMockTest {
+
+    @RegisterExtension
+    static final QuarkusComponentTest test = new QuarkusComponentTest(ProgrammaticLookComponent.class);
```

Comment:
I wonder if we can also provide an annotation based version for simple cases.

Reply:
I was thinking the same as the field pattern is definitely not that well known.

Reply:
> I was thinking the same as the field pattern is definitely not that well known.

Back in JUnit4 days it was the only pattern for extensions AFAIK :-)

Yes, we could definitely add the `@QuarkusComponentTest` annotation and rename `QuarkusComponentTest` to something like `QuarkusComponentTestExtension`...

Reply:
> Back in JUnit4 days it was the only pattern for extensions AFAIK :-)

People that used Spring with JUnit 4 in most cases did not come across this

Reply:
Done - see https://github.com/quarkusio/quarkus/pull/33607/files#diff-aaea73c919f71b81abe28b9d15fd0b9c431d68613b2f933b80de74f418998624.